File: leetcode/python/1768_Merge_Strings_Alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        ans = ""
        if len(word1) == len(word2):
            for i in range(len(word1)):
                ans += word1[i] + word2[i]
        elif len(word1) < len(word2):
            for i in range(len(word1)):
                ans += word1[i] + word2[i]
            ans += word2[len(word1):]
        else:
            for i in range(len(word2)):
                ans += word1[i] + word2[i]
            ans += word1[len(word2):]
        return ans

File: leetcode/go/1768_MergeStringsAlternately.go
func mergeAlternately(word1 string, word2 string) string {
    ans := ""
    idx := 0
    for {
        if (idx >= len(word1) && idx >= len(word2)) {break}
        if (idx < len(word1)) {ans += string(word1[idx])}
        if (idx < len(word2)) {ans += string(word2[idx])}
        idx++;
    }
    return ans
}

func mergeAlternately(word1 string, word2 string) string {
    strSlice := make([]string, 0, len(word1) + len(word2))
    idx := 0
    for {
        if (idx >= len(word1) && idx >= len(word2)) {break}
        if (idx < len(word1)) {strSlice = append(strSlice, string(word1[idx]))}
        if (idx < len(word2)) {strSlice = append(strSlice, string(word2[idx]))}
        idx++;
    }
    return strings.Join(strSlice, "")
}

func mergeAlternately(word1 string, word2 string) string {
    var b bytes.Buffer
    idx := 0
    for {
        if (idx >= len(word1) && idx >= len(word2)) {break}
        if (idx < len(word1)) {b.WriteString(string(word1[idx]))}
        if (idx < len(word2)) {b.WriteString(string(word2[idx]))}
        idx++;
    }

    return b.String();
}